pilot/cmd/pilot-agent/status/server.go

```diff
@@ -601,6 +615,19 @@ func (s *Server) handleAppProbe(w http.ResponseWriter, req *http.Request) {
 	w.WriteHeader(response.StatusCode)
 }
 
+func (s *Server) handleAppProbeTCPSocket(w http.ResponseWriter, req *http.Request, prober *Prober, path string) {
+	port := prober.TCPSocket.Port.IntValue()
+	timeout := time.Duration(prober.TimeoutSeconds) * time.Second
+
+	conn, err := net.DialTimeout("tcp", fmt.Sprintf("%s:%d", s.appProbersDestination, port), timeout)
+	if err != nil {
+		w.WriteHeader(404)
```

Comment:
```suggestion
		w.WriteHeader(http.StatusInternalServerError)
```

Seems like it may be more appropriate? 404 seems like it would suggest we hit some URL that wasn't configured, rather than the health check failed